contrib/go-redis/redis/redis.go

```diff
@@ -193,7 +191,40 @@ func createWrapperFromClient(tc *Client) func(oldProcess func(cmd redis.Cmder) e
 	}
 }
 
-func cmderToString(cmd redis.Cmder) string {
+func cmderToCmdStrAndLength(cmder redis.Cmder) (string, string, int) {
+	return cmdStrToTagStr(_cmderToString(cmder))
+}
+
+func cmderToString(cmder redis.Cmder) string {
+	raw, _, _ := cmdStrToTagStr(_cmderToString(cmder))
+	return raw
+}
+
+func cmdStrToTagStr(cmdStr string) (string, string, int) {
+	parts := strings.Split(strings.Trim(cmdStr, " "), " ")
+	if len(parts) == 0 {
+		return cmdStr, cmdStr, 0
+	}
+
+	cmd := parts[0]
+	argsLength := len(parts) - 1
+	numArgs, ok := cmdCaptureArguments[cmd]
+	if !ok {
+		return cmdStr, cmd, argsLength
```

Comment:
If I understand this logic correctly, any unrecognized/not-explicitly-encoded command will just have part[0] (the command name and only the command name, no args) as the span name?

Reply:
It's the reverse. Any unrecognized command will use the full command with args as the tag value (preserve current behavior).

Reply:
Perhaps for safety it should be the other way?  The Jedis (Java) instrumentation appears to just use the command name (first part) and nothing else (though I like the idea of porting the "this number of params is safe" there).

Reply:
The list of known commands is quite exhaustive. I actually went over all commands and listed down every command I found that wrote data to the DB or queried data using values that might be in the DB. All other commands either use keys/IDs and never take any "data" as arguments so it's safe to keep them at least from data volume perspective. The downside is that a new "write" command in future would escape this filter but it's probably not that big of a deal given than this PR mainly tries to reduce trace volume.